contracts/core/State.sol

```diff
@@ -160,6 +162,18 @@ contract State is Objects, ReentrancyGuard, Ownable {
 
 	address public admin;
 
+	address public protocolAddress; // for modules interaction
+
+	mapping(address => bool) public userNotFirstTradeFlag; // The flag is set on the user's first trade or borrowing
+
+	mapping(address => address) public affiliatesUserReferrer; // User => referrer (affiliate)
+	mapping(address => EnumerableAddressSet.AddressSet) internal referralsList; // list of referral addresses that owned by the referrer
+
+	uint256 public minReferralsToPayout = 3;
+	mapping(address => uint256) public affiliateRewardsHeld; // Total affiliate rewards that held in the protocol (Because the minimum referrals is less than the rule)
+	address public sovTokenAddress; // For affiliates SOV Bonus proccess
+	address public lockedSOVAddress;
+
```

Comment:
⚠️ just to remind that before merging needs to be synced from the development branch and verified that all new variables are appended to avoid storage memory overlapping ⚠️